sdk/core/azure-core/src/main/java/com/azure/android/core/http/interceptor/LoggingInterceptor.java

```diff
@@ -96,24 +103,42 @@ public Response intercept(@NonNull Chain chain) throws IOException {
      *
      * @param request The HTTP request being sent to Azure.
      */
-    private void logRequest(final Request request) {
+    private void logRequest(final Request request) throws IOException {
         HttpUrl url = request.url();
 
-        logger.info("--> [" + request.header(CLIENT_REQUEST_ID) + "]"); // Request ID
+        logger.info("--> [" + request.header(REQUEST_ID_HEADER) + "]"); // Request ID
         logger.info(request.method() + " " + url.encodedPath() + LogUtils.getRedactedQueryString(url,
             allowedQueryParameterNames)); // URL path + query
         logger.info("Host: " + url.scheme() + "://" + url.host()); // URL host
 
         // TODO: Add log level guard for headers and body.
-        logHeaders(request.headers());
-
-        String bodyEvaluation = LogUtils.evaluateBody(request.headers());
         RequestBody requestBody = request.body();
+        boolean contentTypeLogged = false;
+        boolean contentLengthLogged = false;
+
+        // According to the OkHttp documentation, the request body headers are only present when this is installed as a
+        // network interceptor.
+        // https://github.com/square/okhttp/blob/b189a382bccc1b9a01d4672210b69680d73b4306/okhttp-logging-interceptor/src/main/java/okhttp3/logging/HttpLoggingInterceptor.java#L173
+        if (requestBody != null) {
+            if (requestBody.contentType() != null  && allowedHeaderNames.contains(CONTENT_TYPE_HEADER)) {
```

Comment:
Why do these headers get special treatment?

Reply:
If the Content-Type or Content-Length are not present in the headers, we don't need to "synthesize" them for logging purposes.

The only reason I could see needing to do that is if OkHTTP itself adds those headers, but it hasn't done so yet (by the time our logging interceptor is called). Which is pretty annoying if that's the case, since interceptors should be able to assume the headers list is finalized by the time they receive the request or response. If that's what's going on, we can keep this code but I think we could probably refactor it a bit to make it more clear. I'll poke at this while I have the PR checked out as mentioned above.